### PR TITLE
Categorized list for general and inline feedback

### PR DESCRIPTION
**Pull request by crustypluto19**
- Changed the list styling (now collapsible) and white background
- Add general and inline feedback accordingly

### CODE REVIEW

File: Themis/Views/Assessment/CorrectionSidebar/GeneralFeedbackCellView.swift [hunk 1]
```diff
@@ -21,21 +22,40 @@ struct GeneralFeedbackCellView: View {
...
                     Button {
-                        showAddFeedback = true
+                        feedbackType = .general
+                        showAddFeedback.toggle()
                     } label: {
                         Image(systemName: "plus")
                     }
                 }.padding()) {
-                    ForEach(assessment.feedback.feedbacks) { feedback in
+                    ForEach(assessment.feedback.feedbacks.filter { $0.type == .general }) { feedback in
```
Comment (AlexExcellence) [inline Themis/Views/Assessment/CorrectionSidebar/GeneralFeedbackCellView.swift]: AssessmentModel already contains a computed property that filters the general feedbacks. Use that one instead

File: Themis/Views/Assessment/CorrectionSidebar/GeneralFeedbackCellView.swift [hunk 2]
```diff
@@ -21,21 +22,40 @@ struct GeneralFeedbackCellView: View {
...
+                        .font(.title3)
+                    Spacer()
+                    Button {
+                        feedbackType = .inline
+                        showAddFeedback.toggle()
+                    } label: {
+                        Image(systemName: "plus")
+                    }
+                }.padding()) {
+                    ForEach(assessment.feedback.feedbacks.filter { $0.type == .inline }) { feedback in
```
Comment (AlexExcellence) [inline Themis/Views/Assessment/CorrectionSidebar/GeneralFeedbackCellView.swift]: same thing like above

**Review by AlexExcellence — CHANGES_REQUESTED** [review]

**Review by AlexExcellence — APPROVED** [review]

**Review by AlexExcellence — COMMENTED** [review]
The "+" Button for the inline feedbacks section should be removed since inline feedbacks should always include a code reference so creating them from the correction sidebar doesn't really make sense (Then you also don't need the type property anymore).
We could also reduce the titles of the Sections to "General" and "Inline" since the User can already see in the tabbar that he's in the feedback tab.
Personally I would also set the listRowSeparators to hidden since the items are already visually separated by its borders.

**Comment by botrufus** [comment]
Changed some things while trying around and reviewing the pr. Implemented Alex's feedback and refined some of the feedback cell views design. Take a look at it, if you don't like some changes feel free to change it back or even revert the commit.

**Review by botrufus — APPROVED** [review]

**Pull request merged**